File: common/src/dtos/amqp/ticker-event-message.ts
import { TickerEventType } from "../../common/enums";

export class TickerEventMessage {
  readonly time: Date;
  readonly pair: string;
  readonly price: number;
  readonly type: TickerEventType;
  readonly size: number;
  readonly exchange: string;

  constructor(data: {time: Date, pair: string, price: number, type: TickerEventType, size: number, exchange: string} ) {
    this.time = data.time;
    this.pair = data.pair;
    this.price = data.price;
    this.type = data.type;
    this.size = data.size;
    this.exchange = data.exchange;
  }
}

File: message-listener-java/common/src/main/java/com/ticker/message/listener/common/dtos/kafka/TickerEventMessage.java
package com.ticker.message.listener.common.dtos.kafka;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.ticker.message.listener.common.enums.TickerEventType;
import lombok.Data;

import java.util.Date;

@Data
@JsonIgnoreProperties(ignoreUnknown = true)
public class TickerEventMessage {
    private Date time;
    private String pair;
    private Double price;
    private TickerEventType type;
    private Double size;
    private String exchange;
}
